### DISCUSSION TITLE
Trying to use string constant as store name/id

### DISCUSSION THREAD

**Discussion by ayochim** [vuejs/pinia#2739] [Help and Questions] [answered]
Is this possible?
I have an object of string constants exported like so:
```typescript
export const StoreName = {
  StoreOne: 'storeOne',
  StoreTwo: 'storeTwo',
  StoreThree: 'storeThree'
} as const
```

Then I want to use it like so
```typescript
export default defineStore(StoreName.StoreOne, {
  ...
  ...
}
```

But I keep getting errors that StoreName is undefined.

My end goal is to keep a map of store name / store state (dirty, etc) so I can better manage my states, and wanted the names to be consistent throughout the app.

**Comment by posva** [comment, marked as answer]
You can but there is no point unless you use check the store's id somewhere. You might be missing the import of `StoreName`

**Reply by ayochim** [reply]
Thanks for your response. I am confident I had all the imports. I never did figure out the root cause of this but ended up going in a totally different direction.
